### ISSUE TITLE
Robot placing bug with RFTools

### ISSUE THREAD

**Issue by lkoelln** [MightyPirates/OpenComputers#2224]
So I'm new to OpenComputers but I think I just found a bug using the place() method of a robot.
I'm using OC together with RFTools and I want my robot to pickup and replace some Endergenic Generator blocks. But:
The robot ist not able to correctly place this block in the world. I'm selecting the correct slot, but trying to place it just returns an "unknown error".
And now the interesting part: If the robot is equipped with a RFTools "Smart Wrench" it will place an Endergenic Generator but will use up the wrench and not the generator block. So basically it is replacing the wrench with a new generator block.
After the wrench is used up, it won't place one again until I put a new wrench into the equipment slot of the robot.

I'm using the current 1.6 HermitPack. Minecraft 1.10.2 and OC 1.10.2-1.6.1.6

I hope this can be fixed :)

Edit:
Just found a stacktrace thats happening if I'm trying to place the generator without a wrench:

```
java.lang.NullPointerException
	at net.minecraft.item.ItemBlock.func_180614_a(ItemBlock.java:49) ~[acl.class:?]
	at mcjty.lib.compat.CompatItemBlock.clOnItemUse(CompatItemBlock.java:39) ~[CompatItemBlock.class:?]
	at mcjty.lib.compat.CompatItemBlock.func_180614_a(CompatItemBlock.java:35) ~[CompatItemBlock.class:?]
	at net.minecraftforge.common.ForgeHooks.onPlaceItemIntoWorld(ForgeHooks.java:780) ~[ForgeHooks.class:?]
	at net.minecraft.item.ItemStack.func_179546_a(ItemStack.java:144) ~[adz.class:?]
	at li.cil.oc.server.agent.Player.li$cil$oc$server$agent$Player$$tryPlaceBlockWhileHandlingFunnySpecialCases(Player.scala:475) ~[Player.class:?]
	at li.cil.oc.server.agent.Player$$anonfun$placeBlock$1.apply(Player.scala:292) ~[Player$$anonfun$placeBlock$1.class:?]
	at li.cil.oc.server.agent.Player$$anonfun$placeBlock$1.apply(Player.scala:287) ~[Player$$anonfun$placeBlock$1.class:?]
	at li.cil.oc.server.agent.Player.callUsingItemInSlot(Player.scala:438) ~[Player.class:?]
	at li.cil.oc.server.agent.Player.placeBlock(Player.scala:287) ~[Player.class:?]
	at li.cil.oc.server.component.Agent$$anonfun$place$1.apply(Agent.scala:261) ~[Agent$$anonfun$place$1.class:?]
	at li.cil.oc.server.component.Agent$$anonfun$place$1.apply(Agent.scala:252) ~[Agent$$anonfun$place$1.class:?]
	at scala.collection.immutable.List.foreach(List.scala:383) ~[List.class:?]
	at li.cil.oc.server.component.Agent$class.place(Agent.scala:252) ~[Agent$class.class:?]
	at li.cil.oc.server.component.Robot.place(Robot.scala:29) ~[Robot.class:?]
	at generated.li.cil.oc.CallWrapper_li_cil_oc_server_component_Robot_place.call(Unknown Source) ~[?:?]
	at li.cil.oc.server.machine.Callbacks$ComponentCallback.apply(Callbacks.scala:123) ~[Callbacks$ComponentCallback.class:?]
	at li.cil.oc.server.network.Component$class.invoke(Component.scala:112) ~[Component$class.class:?]
	at li.cil.oc.server.network.Network$ComponentConnectorBuilder$$anon$1.invoke(Network.scala:623) ~[Network$ComponentConnectorBuilder$$anon$1.class:?]
	at li.cil.oc.server.machine.Machine.invoke(Machine.scala:358) ~[Machine.class:?]
	at li.cil.oc.server.machine.luac.ComponentAPI$$anonfun$initialize$5$$anonfun$apply$6.apply(ComponentAPI.scala:78) ~[ComponentAPI$$anonfun$initialize$5$$anonfun$apply$6.class:?]
	at li.cil.oc.server.machine.luac.ComponentAPI$$anonfun$initialize$5$$anonfun$apply$6.apply(ComponentAPI.scala:78) ~[ComponentAPI$$anonfun$initialize$5$$anonfun$apply$6.class:?]
	at li.cil.oc.server.machine.luac.NativeLuaArchitecture.invoke(NativeLuaArchitecture.scala:55) [NativeLuaArchitecture.class:?]
	at li.cil.oc.server.machine.luac.ComponentAPI$$anonfun$initialize$5.apply(ComponentAPI.scala:78) [ComponentAPI$$anonfun$initialize$5.class:?]
	at li.cil.oc.server.machine.luac.ComponentAPI$$anonfun$initialize$5.apply(ComponentAPI.scala:74) [ComponentAPI$$anonfun$initialize$5.class:?]
	at li.cil.oc.util.ExtendedLuaState$ExtendedLuaState$$anon$1.invoke(ExtendedLuaState.scala:24) [ExtendedLuaState$ExtendedLuaState$$anon$1.class:?]
	at li.cil.repack.com.naef.jnlua.LuaState.lua_pcall(Native Method) [LuaState.class:?]
	at li.cil.repack.com.naef.jnlua.LuaState.call(LuaState.java:677) [LuaState.class:?]
	at li.cil.oc.server.machine.luac.NativeLuaArchitecture.runSynchronized(NativeLuaArchitecture.scala:177) [NativeLuaArchitecture.class:?]
	at li.cil.oc.server.machine.Machine.update(Machine.scala:562) [Machine.class:?]
	at li.cil.oc.common.EventHandler$$anonfun$onServerTick$2.apply(EventHandler.scala:177) [EventHandler$$anonfun$onServerTick$2.class:?]
	at li.cil.oc.common.EventHandler$$anonfun$onServerTick$2.apply(EventHandler.scala:175) [EventHandler$$anonfun$onServerTick$2.class:?]
	at scala.collection.mutable.HashSet.foreach(HashSet.scala:78) [HashSet.class:?]
	at li.cil.oc.common.EventHandler$.onServerTick(EventHandler.scala:175) [EventHandler$.class:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_547_EventHandler$_onServerTick_ServerTickEvent.invoke(.dynamic) [?:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) [ASMEventHandler.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185) [EventBus.class:?]
	at net.minecraftforge.fml.common.FMLCommonHandler.onPreServerTick(FMLCommonHandler.java:274) [FMLCommonHandler.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:602) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:471) [MinecraftServer.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_91]
```

**Comment by lkoelln** [comment]
Fancy, didn't know github does cross-project referencing.
I just also posted the problem on the RFTools page, because I'm not sure on whch side this problem lies. I didn't had problems with blocks from other mods.
I also found out, that it does not need to be a Smart Wrench in the equipment slot. Cobblestone can also be used to produce endless Generator or some other RFTools blocks.

Also I tried to let the robot use the wrench to set the destination of the generator blocks. It sadly seems not to work. The robot gives me a succesful return and "block activated" on both blocks, but it does not seem to really connect them.
